sample_project/Assets/SampleViewer/Samples/TimeOfDay/Scripts/SkyControllerComponent.cs

```diff
@@ -0,0 +1,387 @@
+        var speedValue = speed * 100;
+        speedText.text = String.Format("{0:0.#}", speedValue);
+        RotateSky();
+
+        lampPosts = FindObjectOfType<LampPostFeatureQuery>().FeatureItems;
+
+        foreach (var lampPost in lampPosts)
+        {
+            lampPost.GetComponentInChildren<Light>().enabled = time > sunSet || time < sunRise;
+        }
+
+        if (militaryTimeToggle.isOn)
+        {
+            timeMode = TimeMode.MilitaryTime;
+        }
+        else
+        {
+            timeMode = TimeMode.AMPM;
+        }
+
+        timeSlider.onValueChanged.AddListener(delegate
+        {
+            time = timeSlider.value;
+            RotateSky();
+            CalculateTime(time, timeText);
+            ToggleLampPosts();
+        });
+
+        militaryTimeToggle.onValueChanged.AddListener(delegate
+        {
+            if (militaryTimeToggle.isOn)
+            {
+                timeMode = TimeMode.MilitaryTime;
+            }
+            else
+            {
+                timeMode = TimeMode.AMPM;
+            }
+
+            CalculateTime(time, timeText);
+            CalculateTime(startTime, startTimeText);
+            CalculateTime(stopTime, stopTimeText);
+        });
+
+        animateToggle.onValueChanged.AddListener(delegate
+        {
+            if (animateToggle.isOn)
+            {
+                backgroundImage.rectTransform.sizeDelta = new Vector2(backgroundImage.rectTransform.sizeDelta.x, 500);
+                backgroundImage.rectTransform.anchoredPosition = new Vector2(backgroundImage.rectTransform.anchoredPosition.x, -321.02f);
+                timeRange.SetActive(true);
+                animateToggle.interactable = false;
+                simulateToggle.isOn = false;
+                simulateToggle.interactable = true;
+            }
+        });
+
+        simulateToggle.onValueChanged.AddListener(delegate
+        {
+            if (simulateToggle.isOn)
+            {
+                timeRange.SetActive(false);
+                backgroundImage.rectTransform.sizeDelta = new Vector2(backgroundImage.rectTransform.sizeDelta.x, 430);
+                backgroundImage.rectTransform.anchoredPosition = new Vector2(backgroundImage.rectTransform.anchoredPosition.x, -286.07f);
+                simulateToggle.interactable = false;
+                animateToggle.isOn = false;
+                animateToggle.interactable = true;
+            }
+        });
+
+        startButton.onClick.AddListener(delegate
+        {
+            if (animateToggle.isOn && skyMode == SkyMode.None)
+            {
+                skyMode = SkyMode.Animated;
+                time = startTime;
+                startButton.GetComponentInChildren<TextMeshProUGUI>().text = "Stop";
+                simulateToggle.interactable = false;
+                animateToggle.interactable = false;
+                timeSlider.interactable = false;
+                InvokeRepeating(nameof(ChangeMode), 0, 0.01f);
```

Comment:
```suggestion
                InvokeRepeating(nameof(ChangeMode), 0, 0.05f);
```
Not a big deal but I think 0.01 is too fast.

Reply:
0.01 acts as Tick/Update?

Reply:
0.1 is update/tick functionality.  If you think it should be slower it should be a serialized variable.

Reply:
It was meant to match the simulation pace to what was the case before using InvokeRepeating